cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java

```diff
@@ -74,6 +85,150 @@
     this.nsStore = nsStore;
   }
 
+  /**
+   * Returns user repository information
+   */
+  @Path("git/repo/info")
+  @GET
+  public void getRepoInfo(HttpRequest request, HttpResponder responder) throws Exception {
+    //check if repo info exists
+    if (userRepo.toString().equals("{}")) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, GSON.toJson("Please add repo information first."));
+    } else {
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(userRepo));
+    }
+  }
+
+  @Path("git/repo/add")
+  @PUT
+  public void addRepoInfo(FullHttpRequest request, HttpResponder responder) throws Exception {
+    //first check if repository info has already been saved
+    if (!userRepo.toString().equals("{}")) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, GSON.toJson("Repo Information is already saved"));
```

Comment:
Generally in CDAP we throw exception from handler and let the `HttpExceptionHandler` to handle the response.